### ISSUE TITLE
Running stylelint.lint in Node.js does not recognize the style rule in vue. 

### ISSUE THREAD

**Issue by vimplus** [stylelint/stylelint#6189]
### What steps are needed to reproduce the bug?

```js
async function run() {
    const config = await stylelint.resolveConfig('.stylelintrc.js');
         
    const res = await stylelint.lint({
        cwd: __dirname,
        config,
        files: ['.css', '.vue'],
    });
    
    console.log(res.results);
}
run();
```

### What Stylelint configuration is needed to reproduce the bug?

Any rule.

### How did you run Stylelint?

`Node.js`

### Which version of Stylelint are you using?

`14.9.1`

### What did you expect to happen?

Correctly identify style rules in vue file.

### What actually happened?

Regardless of whether the style conforms to the rule, it is recognized as an error.

### Does the bug relate to non-standard syntax?

否
